tests/lib/rules/hook-use-state-names.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @fileoverview Ensure symmetric naming of setState hook value and setter variables
+ * @author Duncan Beevers
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/hook-use-state-names');
+const parsers = require('../../helpers/parsers');
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module'
+  }
+});
+
+ruleTester.run('hook-set-state-names', rule, {
+  valid: [
+    {
+      code: 'const [color, setColor] = useState()'
+    },
+    {
+      code: 'const [color, setColor] = useState(\'#ffffff\')'
+    },
+    {
+      code: 'const [color, setColor] = React.useState()'
+    },
+    {
+      code: 'const [color, setColor, unused] = useState()'
+    },
+    {
+      code: 'const [color1, setColor1] = useState()'
+    },
+    {
+      code: 'const [, setColor] = useState()'
+    },
+    {
+      code: 'const useStateResult = useState()'
+    },
+    {
+      code: 'const [, , unused] = useState()'
```

Comment:
this seems like it should be an error, but i guess not from this rule

Reply:
Yeah, in my workflow this issue would be caught by the type system.
This test is intended to indicate the rule's non-opinion around a third argument (and to make sure I didn't throw an error on weird inputs)

Reply:
This is now an error.